### DISCUSSION TITLE
Adding chips for quality onto posters

### DISCUSSION THREAD

**Discussion by EffakT** [jellyfin/jellyfin-vue#2164] [Features]
When you have a library per quality (quite a lot of people do), e.g "Shows", "Shows - 4K", "Movies", "Movies - 4K"

Currently when searching, or viewing a grid of posters, it is difficult to tell what is 4k, HQ, LQ, HDR, etc.
They currently show as posters side-by-side, with no difference (You have to navigate into the result to see the quality). This is mainly an issue on the search page.
I am aware this is the same as on Jellyfin Web, though could possibly be done there aswell?

My thoughts are to add chips/badges for "4k", "HQ", "LQ" next to the watched/unwatched chip, if there are multiple results (or even if not).

**Comment by ferferga** [comment]
Not really answering your question but, is there any good reason why do you have them set up in different libraries? And why other people do?

Do you know an item can have multiple versions? Jellyfin should also support videos with multiple streams, though I don't think that's done here in Vue.

I don't really hearing this is a common practice, the few on the top of my head were people who precisely didn't know about the multiple versions feature.

**Comment by EffakT** [comment]
Yep, I'm aware of the multiple versions feature, however one of the reason I use multiple libraries is that I have some users who don't support 4K, but are also not very tech-savvy.
This means they might try playing a 4K video on a 1080p TV without HDR support.
I'd rather not have my server needlessly transcoding 4K video when I have a 1080p video available.
From what I could tell, the users would have to select their version (I might be wrong about that)?

There are also current restrictions in applications like Sonarr/Radarr where it pulls only the best quality, and overrides the old. This means if you request 1080P and 4K, as soon as 4K is available it deletes the 1080p version.
The only workaround for this is a separate folder for 1080P and 4K.
I am aware that Radarr is working on supporting multiple versions for their v6 release.

I have attempted in the past to setup a library with both 1080p and 4K folders, however the multiple-versions did not work and added the movie/series twice. From what I'm aware, Jellyfin requires the versions to be side-by-side in a single folder. E.G:

This works:
/Movies/Break of Dawn (2022)/Break of Dawn (2022) - 1080p.mp4
/Movies/Break of Dawn (2022)/Break of Dawn (2022) - 4K.mp4

This doesn't:
/Movies/Break of Dawn (2022)/Break of Dawn (2022) - 1080p.mp4
/Movies-4K/Break of Dawn (2022)/Break of Dawn (2022) - 4K.mp4


This is probably not the best place to go into depth with issues related to multiple-versions though, as there are already issues in the main repo of Jellyfin regarding this specific issue.

**Reply by EffakT** [reply]
There is actually a solution to this using the Multiple Versions plugin, however there are still limitations.
E.G one solution was to hide the 4K libraries from users, and use merge to merge these into the HD versions. Limitation here is that you require a 1080p file to view the movie/series.

This is why I was hoping for adding chips/badges to the posters. This could still support multiple versions by adding 4K, HD, 3D badges to the posters where applicable so you could technically have multiple badges. The user could turn these "quality" badges on/off as required if they are needed or not.
